projects/knora-ui/src/lib/viewer/values/time-value/time-input/time-input.component.ts

```diff
@@ -0,0 +1,236 @@
+    const isSubmitted = form && form.submitted;
+    return !!(control && control.invalid && (control.dirty || control.touched || isSubmitted));
+  }
+}
+
+class MatInputBase {
+  constructor(public _defaultErrorStateMatcher: ErrorStateMatcher,
+              public _parentForm: NgForm,
+              public _parentFormGroup: FormGroupDirective,
+              public ngControl: NgControl) {}
+}
+const _MatInputMixinBase: CanUpdateErrorStateCtor & typeof MatInputBase = mixinErrorState(MatInputBase);
+
+export class DateTime {
+  /**
+   * @param date DateTime's date.
+   * @param time DateTime's time.
+   */
+  constructor(public date: GregorianCalendarDate, public time: string) {
+  }
+}
+
+@Component({
+  selector: 'kui-time-input',
+  templateUrl: './time-input.component.html',
+  styleUrls: ['./time-input.component.scss'],
+  providers: [{provide: MatFormFieldControl, useExisting: TimeInputComponent}]
+})
+export class TimeInputComponent extends _MatInputMixinBase implements ControlValueAccessor, MatFormFieldControl<string>, DoCheck, CanUpdateErrorState, OnDestroy{
+
+  static nextId = 0;
+
+  form: FormGroup;
+  stateChanges = new Subject<void>();
+  @HostBinding() id = `kui-time-input-${TimeInputComponent.nextId++}`;
+  focused = false;
+  errorState = false;
+  controlType = 'kui-time-input';
+  matcher = new TimeInputErrorStateMatcher();
+  onChange = (_: any) => {};
+  onTouched = () => {};
+
+  @Input() dateLabel = 'date';
+  @Input() timeLabel = 'time';
+
+  dateFormControl: FormControl;
+  timeFormControl: FormControl;
+
+  datePipe = new DatePipe('en-US');
+
+  get empty() {
+    const userInput = this.form.value;
+    return !userInput.date && !userInput.time;
+  }
+
+  @HostBinding('class.floating')
+  get shouldLabelFloat() {
+    return this.focused || !this.empty;
+  }
+
+  @Input()
+  get required() {
+    return this._required;
+  }
+
+  set required(req) {
+    this._required = coerceBooleanProperty(req);
+    this.stateChanges.next();
+  }
+  
+  private _required = false;
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+    this._disabled ? this.form.disable() : this.form.enable();
+    this.stateChanges.next();
+  }
+
+  private _disabled = false;
+
+  @Input()
+  get placeholder() {
+    return this._placeholder;
+  }
+
+  set placeholder(plh) {
+    this._placeholder = plh;
+    this.stateChanges.next();
+  }
+
+  private _placeholder: string;
+
+  @Input() readonly = false;
+
+  @HostBinding('attr.aria-describedby') describedBy = '';
+
+  setDescribedByIds(ids: string[]) {
+    this.describedBy = ids.join(' ');
+  }
+
+  @Input()
+  get value(): string | null {
+    const userInput = new DateTime(this.form.value.date, this.form.value.time);
+    if (userInput.date !== null && userInput.time !== null && userInput.time !== '') {
```

Comment:
I think you should also check the time string against the regex here (to prevent an error in `this.userInputToTimestamp `). If invalid, `null` is returned.

Reply:
or probably you can just see if the `FormControl` for time is valid

Reply:
fixed in 548005d